src/UnoFlipController.java

```diff
@@ -86,42 +105,62 @@ public void actionPerformed(ActionEvent e) {
                         null,
                         null);
 
-                Card.Colour colour;
-                if (result == JOptionPane.OK_OPTION) {
+                // Colour is selected and OK is pressed
```

Comment:
Move this comment inside the if block (because this comment is for IF result == JOptionPane.OK_OPTION. Also, you can provide general explanation as to what's happening in the if-block instead of breaking down every line. Especially the redundant comments like "Sets the new current colour" when the line of code is this.model.setCurrentColour(colour).